### ISSUE TITLE
starts with bam file

### ISSUE THREAD

**Issue by huangl07** [hsinnan75/MapCaller#27]
My dear

I have already test your software，it was extremely faster，could it start with bam files

thankyou!

**Comment by hsinnan75** [comment]
Hello,
Thank you for the suggestion. The reason we developed MapCaller is to integrate read mapping and variant calling into a single process for saving running time and disk space. We'll investigate how will MapCaller perform when it takes BAM/SAM files as input.

**Comment by huangl07** [comment]
well,I find that the MapCaller mapping result didn't fit bwa well, I find the mapping ratio is lower than bwa

**Comment by hsinnan75** [comment]
That's true. According to our analysis, BWA produces higher sensitivity for read mapping. However, you don't have to map every read for variant calling. That's why we adopted a more efficient way to call variants. MapCaller discards read alignments whose number of mismatches > 5, and considers them unmapped. That explains why MapCaller produces lower sensitivity. Since the major purpose of MapCaller is to call variants, we only consider desirable read alignments.

**Comment by tseemann** [comment]
>  MapCaller discards read alignments whose number of mismatches > 5, and considers them unmapped.

Can you make that a parameter?
eg. `-maxmm 5` ?  or `-maxmmpc 5` for % of read length?

**Comment by hsinnan75** [comment]
Thanks for the suggestion. I've uploaded the updated version (0.9.9.16). You can specify the maximal mismatch number allowed in a read alignment with -maxmm.
